### PR TITLE
KDESKTOP-1056-Test-to-be-added-for-KDESKTOP-393

### PR DESCRIPTION
**Pull request by ClementKunz**

### CODE REVIEW

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.h [hunk 1]
```diff
@@ -43,10 +45,13 @@ class TestPlatformInconsistencyCheckerWorker : public CppUnit::TestFixture {
         void testCheckNameForbiddenChars();
         void testCheckReservedNames();
         void testNameClash();
+        void testNameClashAfterRename();
         void testExecute();
 
     private:
         std::shared_ptr<SyncPal> _syncPal{nullptr};
+
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.h:53]:
```suggestion
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 2]
```diff
@@ -154,6 +165,78 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    DbNodeId dbNodeIdDir_a;
+    bool constraintError = false;
+    _syncPal->_syncDb->insertNode(dbNode_A, dbNodeIdDir_A, constraintError);
+    _syncPal->_syncDb->insertNode(dbNode_a, dbNodeIdDir_a, constraintError);
+
+    // Set up remote tree
+    std::shared_ptr<Node> remoteParentNode = _syncPal->_remoteUpdateTree->rootNode();
+    std::shared_ptr<Node> node_ra =
+        std::make_shared<Node>(dbNodeIdDir_a, _syncPal->_remoteUpdateTree->side(), Str("a"), NodeTypeFile, OperationTypeMove,
+                               "ra", 0, 0, 12345, _syncPal->_remoteUpdateTree->rootNode());
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:192]:
```suggestion
    std::shared_ptr<Node> node_ra =
        std::make_shared<Node>(dbNodeIdDir_a, ReplicaSideRemote, Str("a"), NodeTypeFile, OperationTypeMove,
                               "ra", 0, 0, 12345, _syncPal->getUpdateTree(ReplicaSideRemote)->rootNode());
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 3]
```diff
@@ -154,6 +165,78 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    std::shared_ptr<Node> node_rA =
+        std::make_shared<Node>(dbNodeIdDir_A, _syncPal->_remoteUpdateTree->side(), Str("A"), NodeTypeFile, OperationTypeNone,
+                               "rA", 0, 0, 12345, _syncPal->_remoteUpdateTree->rootNode());
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:195]:
```suggestion
    std::shared_ptr<Node> node_rA =
        std::make_shared<Node>(dbNodeIdDir_A, ReplicaSideRemote, Str("A"), NodeTypeFile, OperationTypeNone,
                               "rA", 0, 0, 12345, _syncPal->getUpdateTree(ReplicaSideRemote)->rootNode());
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 4]
```diff
@@ -154,6 +165,78 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+
+    _syncPal->_remoteUpdateTree->insertNode(node_ra);
+    _syncPal->_remoteUpdateTree->insertNode(node_rA);
+
+    // Set up local tree
+    std::shared_ptr<Node> localParentNode = _syncPal->_localUpdateTree->rootNode();
+    std::shared_ptr<Node> node_la = std::make_shared<Node>(dbNodeIdDir_a, _syncPal->_localUpdateTree->side(), Str("a1"),
+                                                           NodeTypeFile, OperationTypeNone, "la", 0, 0, 12345, localParentNode);
+    std::shared_ptr<Node> node_lA = std::make_shared<Node>(dbNodeIdDir_A, _syncPal->_localUpdateTree->side(), Str("A"),
+                                                           NodeTypeFile, OperationTypeNone, "lA", 0, 0, 12345, localParentNode);
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:208]:
```suggestion
    std::shared_ptr<Node> node_la = std::make_shared<Node>(dbNodeIdDir_a, ReplicaSideLocal, Str("a1"),
                                                           NodeTypeFile, OperationTypeNone, "la", 0, 0, 12345, localParentNode);
    std::shared_ptr<Node> node_lA = std::make_shared<Node>(dbNodeIdDir_A,ReplicaSideLocal, Str("A"),
                                                           NodeTypeFile, OperationTypeNone, "lA", 0, 0, 12345, localParentNode);
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 5]
```diff
@@ -154,6 +165,78 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+
+    localParentNode->insertChildren(node_la);
+    localParentNode->insertChildren(node_lA);
+
+    _syncPal->_localUpdateTree->insertNode(node_la);
+    _syncPal->_localUpdateTree->insertNode(node_rA);
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:214]:
```suggestion
    _syncPal->getUpdateTree(ReplicaSideLocal)->insertNode(node_la);
    _syncPal->getUpdateTree(ReplicaSideLocal)->insertNode(node_rA);
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 6]
```diff
@@ -154,6 +165,78 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+                               "ra", 0, 0, 12345, _syncPal->_remoteUpdateTree->rootNode());
+    std::shared_ptr<Node> node_rA =
+        std::make_shared<Node>(dbNodeIdDir_A, _syncPal->_remoteUpdateTree->side(), Str("A"), NodeTypeFile, OperationTypeNone,
+                               "rA", 0, 0, 12345, _syncPal->_remoteUpdateTree->rootNode());
+
+    remoteParentNode->insertChildren(node_ra);
+    remoteParentNode->insertChildren(node_rA);
+
+    _syncPal->_remoteUpdateTree->insertNode(node_ra);
+    _syncPal->_remoteUpdateTree->insertNode(node_rA);
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:201]:
```suggestion
    _syncPal->getUpdateTree(ReplicaSideRemote)->insertNode(node_ra);
    _syncPal->getUpdateTree(ReplicaSideRemote)->insertNode(node_rA);
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 7]
```diff
@@ -154,6 +165,78 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    DbNode dbNode_a(3, _syncPal->_syncDb->rootNode().nodeId(), Str("a1"), Str("a1"), "la", "ra", defaultTime, defaultTime,
+                    defaultTime, NodeType::NodeTypeDirectory, defaultSize, std::nullopt);
+    DbNodeId dbNodeIdDir_A;
+    DbNodeId dbNodeIdDir_a;
+    bool constraintError = false;
+    _syncPal->_syncDb->insertNode(dbNode_A, dbNodeIdDir_A, constraintError);
+    _syncPal->_syncDb->insertNode(dbNode_a, dbNodeIdDir_a, constraintError);
+
+    // Set up remote tree
+    std::shared_ptr<Node> remoteParentNode = _syncPal->_remoteUpdateTree->rootNode();
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:189]:
```suggestion
    std::shared_ptr<Node> remoteParentNode = _syncPal->getUpdateTree(ReplicaSideRemote)->rootNode();
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 8]
```diff
@@ -154,6 +165,78 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+        ofs << "Some content.\n";
+        ofs.close();
+    }
+
+    // Set up DB
+    DbNode dbNode_A(2, _syncPal->_syncDb->rootNode().nodeId(), Str("A"), Str("A"), "lA", "rA", defaultTime, defaultTime,
+                    defaultTime, NodeType::NodeTypeDirectory, defaultSize, std::nullopt);
+    DbNode dbNode_a(3, _syncPal->_syncDb->rootNode().nodeId(), Str("a1"), Str("a1"), "la", "ra", defaultTime, defaultTime,
+                    defaultTime, NodeType::NodeTypeDirectory, defaultSize, std::nullopt);
+    DbNodeId dbNodeIdDir_A;
```
Comment (luc-guyot-infomaniak) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:182]: We usually avoid underscores and write variable names with camelCase.

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 9]
```diff
@@ -154,6 +165,78 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    // Create local files
+    std::vector<std::string> nodes = {{"a1"}, {"A"}};
+    for (const auto &n : nodes) {
+        std::ofstream ofs(_tempDir.path / n);
+        ofs << "Some content.\n";
+        ofs.close();
+    }
+
+    // Set up DB
+    DbNode dbNode_A(2, _syncPal->_syncDb->rootNode().nodeId(), Str("A"), Str("A"), "lA", "rA", defaultTime, defaultTime,
```
Comment (luc-guyot-infomaniak) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:178]: Not sure if it is merged yet, but there might be a `const` public accessor to `_syncDb`.

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 10]
```diff
@@ -42,16 +43,26 @@ namespace KDC {
...
+    Drive drive(1, 1, account.dbId(), std::string(), 0, std::string());
+    ParmsDb::instance()->insertDrive(drive);
+
+    Sync sync(1, drive.dbId(), _tempDir.path, "");
+    ParmsDb::instance()->insertSync(sync);
+
+    // Create SyncPal
+    _syncPal = std::make_shared<SyncPal>(sync.dbId(), "3.4.0");
     _syncPal->_syncDb->setAutoDelete(true);
     _syncPal->_tmpBlacklistManager = std::make_shared<TmpBlacklistManager>(_syncPal);
```
Comment (luc-guyot-infomaniak) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:67]: Is `_tmpBlacklistManager ` used by the tests? If not, you may instantiate a `TestSync` object instead. (The manager could also have its own unit tests).

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp (hunk 7 above)
Comment (luc-guyot-infomaniak) [reply test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp]: Is it `getUpdateTree` or simply `updateTree`?

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp (hunk 10 above)
Comment (ClementKunz) [reply test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:67]: Yes, it is used by

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp (hunk 8 above)
Comment (ClementKunz) [reply test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp]: Done

**Review by ChristopheLarchier — APPROVED** [review]

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 11]
```diff
@@ -133,17 +144,17 @@ void TestPlatformInconsistencyCheckerWorker::testCheckReservedNames() {
 }
 
 void TestPlatformInconsistencyCheckerWorker::testNameClash() {
-    std::shared_ptr<Node> parentNode =
+    const auto parentNode =
         std::make_shared<Node>(std::nullopt, _syncPal->_remoteUpdateTree->side(), Str("parentNode"), NodeTypeDirectory,
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:148]:
```suggestion
        std::make_shared<Node>(std::nullopt, _syncPal->updateTree(ReplicaSideRemote)->side(), Str("parentNode"), NodeTypeDirectory,
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 12]
```diff
@@ -133,17 +144,17 @@ void TestPlatformInconsistencyCheckerWorker::testCheckReservedNames() {
...
                                OperationTypeCreate, "parentID", 0, 0, 12345, _syncPal->_remoteUpdateTree->rootNode());
 
-    std::shared_ptr<Node> node_a = std::make_shared<Node>(std::nullopt, _syncPal->_remoteUpdateTree->side(), Str("a"),
-                                                          NodeTypeFile, OperationTypeCreate, "a", 0, 0, 12345, parentNode);
-    std::shared_ptr<Node> node_A = std::make_shared<Node>(std::nullopt, _syncPal->_remoteUpdateTree->side(), Str("A"),
-                                                          NodeTypeFile, OperationTypeCreate, "A", 0, 0, 12345, parentNode);
+    const auto nodeLower = std::make_shared<Node>(std::nullopt, _syncPal->_remoteUpdateTree->side(), Str("a"), NodeTypeFile,
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:151]:
```suggestion
    const auto nodeLower = std::make_shared<Node>(std::nullopt, _syncPal->updateTree(ReplicaSideRemote)->side(), Str("a"), NodeTypeFile,
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 13]
```diff
@@ -133,17 +144,17 @@ void TestPlatformInconsistencyCheckerWorker::testCheckReservedNames() {
...
+                                                  OperationTypeCreate, "a", 0, 0, 12345, parentNode);
+    const auto nodeUpper = std::make_shared<Node>(std::nullopt, _syncPal->_remoteUpdateTree->side(), Str("A"), NodeTypeFile,
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:153]:
```suggestion
    const auto nodeUpper = std::make_shared<Node>(std::nullopt, _syncPal->updateTree(ReplicaSideRemote)->side(), Str("A"), NodeTypeFile,
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 14]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    const DbNode dbNodeUpper(3, _syncPal->_syncDb->rootNode().nodeId(), Str("A"), Str("A"), "lA", "rA", defaultTime, defaultTime,
+                             defaultTime, NodeType::NodeTypeDirectory, defaultSize, std::nullopt);
+    DbNodeId dbNodeIdLower;
+    DbNodeId dbNodeIdUpper;
+    bool constraintError = false;
+    _syncPal->_syncDb->insertNode(dbNodeLower, dbNodeIdLower, constraintError);
+    _syncPal->_syncDb->insertNode(dbNodeUpper, dbNodeIdUpper, constraintError);
+
+    // Set up remote tree
+    const auto remoteParentNode = _syncPal->getUpdateTree(ReplicaSideRemote)->rootNode();
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:188]: `getUpdateTree` is removed in an upcomming PR as it duplicate `updtaeTree`
```suggestion
    const auto remoteParentNode = _syncPal->updateTree(ReplicaSideRemote)->rootNode();
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 15]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    const auto remoteNodeLower =
+        std::make_shared<Node>(dbNodeIdLower, ReplicaSideRemote, Str("a"), NodeTypeFile, OperationTypeMove, "ra", 0, 0, 12345,
+                               _syncPal->getUpdateTree(ReplicaSideRemote)->rootNode());
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:191]:
```suggestion
                               _syncPal->updateTree(ReplicaSideRemote)->rootNode());
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 16]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    const auto remoteNodeUpper =
+        std::make_shared<Node>(dbNodeIdUpper, ReplicaSideRemote, Str("A"), NodeTypeFile, OperationTypeNone, "rA", 0, 0, 12345,
+                               _syncPal->getUpdateTree(ReplicaSideRemote)->rootNode());
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:194]:
```suggestion
                               _syncPal->updateTree(ReplicaSideRemote)->rootNode());
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 17]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+
+    remoteParentNode->insertChildren(remoteNodeLower);
+    remoteParentNode->insertChildren(remoteNodeUpper);
+
+    _syncPal->getUpdateTree(ReplicaSideRemote)->insertNode(remoteNodeLower);
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:199]:
```suggestion
    _syncPal->updateTree(ReplicaSideRemote)->insertNode(remoteNodeLower);
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 18]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    _syncPal->getUpdateTree(ReplicaSideRemote)->insertNode(remoteNodeUpper);
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:200]:
```suggestion
    _syncPal->updateTree(ReplicaSideRemote)->insertNode(remoteNodeUpper);
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 19]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+
+    // Set up local tree
+    const auto localParentNode = _syncPal->getUpdateTree(ReplicaSideLocal)->rootNode();
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:203]:
```suggestion
    const auto localParentNode = _syncPal->updateTree(ReplicaSideLocal)->rootNode();
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 20]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    const auto localNodeLower = std::make_shared<Node>(dbNodeIdLower, ReplicaSideLocal, Str("a1"), NodeTypeFile,
+                                                       OperationTypeNone, "la", 0, 0, 12345, localParentNode);
+    const auto localNodeUpper = std::make_shared<Node>(dbNodeIdUpper, ReplicaSideLocal, Str("A"), NodeTypeFile, OperationTypeNone,
+                                                       "lA", 0, 0, 12345, localParentNode);
+
+    localParentNode->insertChildren(localNodeLower);
+    localParentNode->insertChildren(localNodeUpper);
+
+    _syncPal->getUpdateTree(ReplicaSideLocal)->insertNode(localNodeLower);
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:212]:
```suggestion
    _syncPal->updateTree(ReplicaSideLocal)->insertNode(localNodeLower);
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 21]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
+    _syncPal->getUpdateTree(ReplicaSideLocal)->insertNode(localNodeUpper);
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:213]:
```suggestion
    _syncPal->updateTree(ReplicaSideLocal)->insertNode(localNodeUpper);
```

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp [hunk 22]
```diff
@@ -154,36 +165,111 @@ void TestPlatformInconsistencyCheckerWorker::testNameClash() {
...
 
-    CPPUNIT_ASSERT(_syncPal->_remoteUpdateTree->exists(*parentNode->id()));
+    CPPUNIT_ASSERT(_syncPal->updateTree(ReplicaSideRemote)->exists(*parentNode->id()));
 
 #if defined(WIN32) || defined(__APPLE__)
     CPPUNIT_ASSERT(exactly1exist);
 #else
-    CPPUNIT_ASSERT(_syncPal->_remoteUpdateTree->exists(*node_A->id()) && _syncPal->_remoteUpdateTree->exists(*node_a->id()));
+    CPPUNIT_ASSERT(_syncPal->_remoteUpdateTree->exists(*nodeUpper->id()) &&
+                   _syncPal->_remoteUpdateTree->exists(*nodeLower->id()));
```
Comment (herve-er) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:272]:
```suggestion
    CPPUNIT_ASSERT(_syncPal->updateTree(ReplicaSideRemote)->exists(*nodeUpper->id()) &&
                   _syncPal->updateTree(ReplicaSideRemote)->exists(*nodeLower->id()));
```

Comment (ClementKunz) [reply test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp]: Done

**Review by herve-er — APPROVED** [review]

**Pull request merged**